spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/messaging/TraceMessagingAutoConfiguration.java

```diff
@@ -366,3 +397,48 @@ private boolean typeMatches(Object bean) {
 	}
 
 }
+
+class SqsQueueMessageHandlerFactory extends QueueMessageHandlerFactory {
+
+	private TracingMethodMessageHandlerAdapter handlerAdapter;
+
+	SqsQueueMessageHandlerFactory(TracingMethodMessageHandlerAdapter handlerAdapter) {
+		this.handlerAdapter = handlerAdapter;
+	}
+
+	@Override
+	public QueueMessageHandler createQueueMessageHandler() {
+		if (CollectionUtils.isEmpty(getMessageConverters())) {
+			return new SqsQueueMessageHandler(handlerAdapter, Collections.emptyList());
+		}
+		else {
+			return new SqsQueueMessageHandler(handlerAdapter, getMessageConverters());
+		}
+	}
+
+}
+
+class SqsQueueMessageHandler extends QueueMessageHandler {
+
+	static final String LOGICAL_RESOURCE_ID = "LogicalResourceId"; // copied from QueueMessageHandler
+
+	private TracingMethodMessageHandlerAdapter handlerAdapter;
+
+	SqsQueueMessageHandler(TracingMethodMessageHandlerAdapter handlerAdapter,
+			List<MessageConverter> messageConverters) {
+		super(messageConverters);
+		this.handlerAdapter = handlerAdapter;
+	}
+
+	@Override
+	public void handleMessage(Message<?> message) throws MessagingException {
+		handlerAdapter.wrapMethodMessageHandler(message, super::handleMessage, this::messageSpanTagger);
+	}
+
+	private void messageSpanTagger(Span span, Message<?> message) {
+		span.remoteServiceName("amazon-sqs");
```

Comment:
nit: is it necessary to prefix `amazon-` There are clones of SQS which possibly work regardless

```suggestion
		span.remoteServiceName("sqs");
```